docs/source/training.rst

```diff
@@ -0,0 +1,372 @@
+      - Plotting script: ``imas_muscle3/visualization/examples/simple_1d_plot/simple_1d_plot.py``
+
+      This plotting script is a simple configuration that will only plot the plasma current 
+      of an equilibrium IDS over time. 
+      Run the MUSCLE pipeline, supplying the yMMSL file you made:
+      
+      .. code-block:: bash
+        
+         muscle_manager --start-all <yMMSL file>
+
+      The visualization actor will automatically open your default browser after it is initiated.
+      What do you see in your browser?
+
+      .. hint::
+         Take a look at the example yMMSL file in ``imas_muscle3/visualization/examples/simple_1d_plot/simple_1d_plot.ymmsl``. If you want detailed information about the visualization actor, take a look
+         at the :ref:`documentation <actor_visualization>`.
+
+   .. md-tab-item:: Solution
+
+      Create a file called ``my_visualization.ymmsl`` with the following content:
+
+      .. code-block:: yaml
+
+         ymmsl_version: v0.1
+         model:
+           name: my_visualization
+           components:
+             source_component:
+               implementation: source_component
+               ports:
+                 o_i: [equilibrium_out]
+             visualization_component:
+               implementation: visualization_component
+               ports:
+                 s: [equilibrium_in]
+           conduits:
+             source_component.equilibrium_out: visualization_component.equilibrium_in
+         settings:
+           source_component.source_uri: imas:hdf5?path=/home/ITER/blokhus/public/imasdb/ITER/4/666666/3/
+           visualization_component.plot_file_path: <path/to/IMAS-MUSCLE3>/imas_muscle3/visualization/examples/simple_1d_plot/simple_1d_plot.py
+         implementations:
+           visualization_component:
+             executable: python
+             args: -u -m imas_muscle3.actors.visualization_component
+           source_component:
+             executable: python
+             args: -u -m imas_muscle3.actors.source_component
+         resources:
+           source_component:
+             threads: 1
+           visualization_component:
+             threads: 1
+
+      When you launch the muscle_manger, the browser should open, and you will see the
+      plasma current plotted over time, updating in real-time as the new time slices are 
+      received by the visualization actor.
+
+      .. figure:: ../source/images/ip_curve.gif
+
+
+Exercise 1b: Understanding the Basic Structure
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. md-tab-set::
+   .. md-tab-item:: Exercise
+
+      Now that you were able to run the visualization actor in the previous exercise, let's
+      take a look under the hood to see what the plotting script actually does.
+
+      Every plotting script for the visualization actor must include the following two classes:
+
+      1. ``State(BaseState)``: This class handles extracting and storing data from incoming IDSs.
```

Comment:
explain that this handles every IDS coming in on the S port one at a time with no knowledge of which IDS